src/main/common/draft/draftStoreClient.ts

```diff
@@ -0,0 +1,80 @@
+import { Draft } from 'app/models/draft'
+import { CoreOptions, RequestAPI } from 'request'
+import { RequestPromise } from 'request-promise-native'
+import moment = require('moment')
+
+export default class DraftStoreClient<T> {
+  private endpointURL: string
+  private serviceAuthToken: string
+  private request: RequestAPI<RequestPromise, CoreOptions, CoreOptions>
+
+  constructor (endpointURL: string,
+               serviceAuthToken: string,
+               request: RequestAPI<RequestPromise, CoreOptions, CoreOptions>) {
+    this.endpointURL = endpointURL
+    this.serviceAuthToken = serviceAuthToken
+    this.request = request
+  }
+
+  find (query: { [key: string]: string }, userAuthToken: string,
+        deserializationFn: (value: any) => T): Promise<Draft<T>[]> {
+
+    const { type, ...qs } = query
```

Comment:
As `type` is not passed in query params, I think it should be passed separately to this method.